### DISCUSSION TITLE
style parent element with prose

### DISCUSSION THREAD

**Discussion by hoangviet1807** [tailwindlabs/tailwindcss#12212] [Help] [answered]
How can I style the p element which wraps img element is flex, and the remaining p is block with prose tailwind CSS?
I've tried something like ```prose-img:prose-p:flex``` but it's not working.
Thanks!

**Comment by alex-krasikau** [comment, marked as answer]
Hey @hoangviet1807!

If I understand your issue correctly, you would like to achieve the following outcome:

```css
/* flex */
<p>
  <img />
  <span>Content</span>
</p>

/* no flex */
<p>Content</p>
```

In this case, the `:has` pseudo-class is the only way to create this CSS selector. In pure CSS, it will look like this: `.prose p:has(img)`. However, Tailwind and the typography plugin don't support the `:has` pseudo-class. Therefore, I would recommend using a custom selector declaration like this: `prose [&_p:has(img)]:flex`.

Please note that while `:has` already has decent support across browsers, it's still relatively new and not available in the latest Firefox versions yet (https://caniuse.com/css-has). I would recommend considering other options, especially if you can modify your markup to include `flex` pragmatically directly into wrapping elements or use sibling selectors instead - https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-sibling-state.

**Reply by hoangviet1807** [reply]
it's working, thank you so much, It saved me a lot of time!!!
